lib/smartcar/base.rb

```diff
@@ -23,11 +23,13 @@ class InvalidParameterValue < StandardError; end
       # @param data [Hash] request body if needed.
       #
       # @return [Hash] The response Json parsed as a hash.
-      define_method verb do |path, data = nil|
+      define_method verb do |path, data = nil, headers = {}|
         response = service.send(verb) do |request|
-          request.headers['Authorization'] = auth_type == BASIC ? "Basic #{token}" : "Bearer #{token}"
-          request.headers['sc-unit-system'] = unit_system if unit_system
-          request.headers['Content-Type'] = 'application/json'
+          request_headers = {}
+          request_headers['Authorization'] = auth_type == BASIC ? "Basic #{token}" : "Bearer #{token}"
+          request_headers['sc-unit-system'] = unit_system if unit_system
+          request_headers['Content-Type'] = 'application/json'
+          request.headers = request_headers.merge(headers)
```

Comment:
- So this is overriding whole of the headers. Which means anything being added by ruby before this is going to get overridden.I double checked to see what header is going to get overridden and the only one was `User-Agent` 
```
   33:           byebug
=> 34:           request.headers = request_headers.merge(headers)
   35:           complete_path = "/v#{version}#{path}"
   36:           if verb == :get
   37:             request.url complete_path, data
   38:           else
(byebug) request.headers
{"User-Agent"=>"Faraday v1.8.0"}
```
- That said, the user agent above itself needs to be modified. So can we manually add the `User-Agent` header that follows the spec here https://www.notion.so/smartcarapi/API-SDK-Specification-856d4ece46b14365a967ed36b521b474#f56217dff1d64302871681c66f44decc .

Let me know if this doesn't make sense.

Reply:
This was actually intentional, because in the specification for the request method, we decided that a customer/user should be able to override any header, including User-Agent ([link to request method spec](https://www.notion.so/smartcarapi/API-SDK-Specification-856d4ece46b14365a967ed36b521b474#ee757c9fbab041549762e0edf70f7def)). However, I didn't notice that the default header is Faraday instead of the one in the API Spec, so I'll manually add the *default* User-Agent header to be `Smartcar/<sdk-version> (<os-platform>; <os-arch>) <language> <runtime-version>`.